src/test/unit/gz.test.js

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import { compress, decompress } from '../../utils/gz';
+
+import { TextEncoder, TextDecoder } from 'util';
+
+beforeAll(function() {
+  if ((window: any).TextEncoder) {
+    throw new Error('A TextEncoder was already on the window object.');
+  }
+  (window: any).TextEncoder = TextEncoder;
+});
+
+afterAll(async function() {
+  delete (window: any).TextEncoder;
+});
+
+describe('utils/gz', function() {
+  it('compresses and decompresses properly', async () => {
+    const clearText = '42';
+    const gzipedData = await compress(clearText);
+    expect(gzipedData.constructor.name).toBe('Uint8Array');
```

Comment:
The Uint8Array's instance used is the one from inside the vm, that's why we can't use `toBeInstanceOf`like the other PR.